sys/parsoid.js

```diff
@@ -455,6 +455,67 @@ class ParsoidService {
         return true;
     }
 
+    getHtml(hyper, req) {
+        if (req.query.sections) {
+            return this.getSections(hyper, req);
+        }
+        if (mwUtil.isNoCacheRequest(req)) {
+            // assume this is an update request
+            return this.getFormat('html', hyper, req);
+        }
+        return mwUtil.getSiteInfo(hyper, req).then((siteinfo) => {
+            if (!siteinfo.extensions['Flagged Revisions']) {
+                // the project does not have the FlaggedRevs extension
+                // enabled so proceed as normal
+                return this.getFormat('html', hyper, req);
+            }
+            const rp = req.params;
+            const queryReq = {
+                uri: new URI([rp.domain, 'sys', 'action', 'query']),
+                method: 'post',
+                body: {
+                    prop: 'flagged',
+                    formatversion: 2
+                }
+            };
+            if (rp.revision) {
+                queryReq.body.revids = rp.revision;
+            } else {
+                queryReq.body.titles = rp.title;
+            }
+            return P.props({
+                html: this.getFormat('html', hyper, req),
+                revinfo: hyper.post(queryReq)
+            }).then((res) => {
+                const html = res.html;
+                const retrev = mwUtil.parseETag(html.headers.etag).rev;
+                const revinfo = res.revinfo.body.items[0];
+                if (!revinfo.flagged || revinfo.flagged.stable_revid === retrev) {
+                    // the revision is not flagged, or the client is
```

Comment:
// the page has no flagged revision...

would be a more accurate way to put it.